frontend/src/metabase/visualizations/components/settings/ChartSettingFieldPicker.jsx

```diff
@@ -69,52 +73,109 @@ const ChartSettingFieldPicker = ({
       seriesKey = keyForSingleSeries(seriesForColumn);
     }
   }
+
+  const disabled =
+    options.length === 0 ||
+    (options.length === 1 && options[0].value === value);
+
   return (
     <ChartSettingFieldPickerRoot
       className={className}
-      disabled={options.length === 1 && options[0].value === value}
       showDragHandle={showDragHandle}
       data-testid="chartsettings-field-picker"
+      bg="bg-white"
+      align="center"
     >
-      {showDragHandle && <GrabberHandle name="grabber" noPointer noMargin />}
-      {showColorPicker && seriesKey && (
-        <FieldPickerColorPicker
-          pillSize="small"
-          value={colors[seriesKey]}
-          onChange={value => {
-            onChangeSeriesColor(seriesKey, value);
-          }}
-        />
-      )}
       <ChartSettingSelect
-        value={value}
+        data-testid="chartsettings-field-picker-select"
+        pl="xs"
+        pr="xs"
+        w="100%"
+        isInitiallyOpen={autoOpenWhenUnset && value === undefined}
         options={options}
+        value={value}
         onChange={onChange}
-        placeholder={t`Select a field`}
+        icon={
+          showDragHandle || (showColorPicker && seriesKey) ? (
+            <Group noWrap spacing="xs" pl="sm" pr="xs">
+              {showDragHandle && (
+                <GrabberHandle
+                  name="grabber"
+                  noMargin
+                  onClick={e => e.stopPropagation()}
+                  c="text-medium"
+                  className={CS.pointerEventsAll}
+                />
+              )}
+              {showColorPicker && seriesKey && (
+                <ChartSettingColorPicker
+                  pillSize="small"
+                  value={colors[seriesKey]}
+                  onChange={value => {
+                    onChangeSeriesColor(seriesKey, value);
+                  }}
+                  className={CS.pointerEventsAll}
+                />
+              )}
+            </Group>
+          ) : null
+        }
         placeholderNoOptions={t`No valid fields`}
-        isInitiallyOpen={autoOpenWhenUnset && value === undefined}
-        hiddenIcons
+        placeholder={t`Select a field`}
+        iconWidth="auto"
+        rightSectionWidth="auto"
+        rightSection={
+          <Group noWrap spacing="sm" p="xs" mr="sm">
+            {!disabled && (
+              <ActionIcon c="text-medium" size="sm" radius="xl" p={0}>
+                <Icon name="chevrondown" />
+              </ActionIcon>
+            )}
+            {menuWidgetInfo && (
+              <ChartSettingActionIcon
+                icon="ellipsis"
+                data-testid={`settings-${value}`}
+                onClick={e => {
+                  onShowWidget(menuWidgetInfo, e);
+                }}
```

Comment:
nit: `ChartSettingActionIcon` now returns an `HTMLElement` not an `Event` per the signature. Alternatively we can pass `event` here instead.

```suggestion
                onClick={element => {
                  onShowWidget(menuWidgetInfo, element);
                }}
```